src/routes/Add.jsx

```diff
@@ -1,13 +1,124 @@
-import { Card, CardHeader } from '@nextui-org/react';
+import { useState } from 'react';
+import {
+  Card,
+  CardBody,
+  Input,
+  CardHeader,
+  Divider,
+  Select,
+  SelectItem,
+  Textarea,
+  Button
+} from '@nextui-org/react';
+import { serverTimestamp } from 'firebase/firestore';
+import LoadingCard from '../pages/Dictionary/components/LoadingCard';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { unpackTermsQuery } from '../utils/unpackQuery';
+import getTermsQuery from '../utils/getTermsQuery';
+import getEntriesMutation from '../utils/getEntriesMutation';
 
 const Add = () => {
+  const [termSelectBox, setTermSelectBox] = useState('');
```

Comment:
NextUI's input, textbox already handles states internally so we don't need to make useState. Checkout Tiana's Input component in UserSignUp.jsx file in this PR https://github.com/rainawan/BruinDictionary/pull/21/files

Reply:
**useRef**
useRef is primarily used to access and manipulate the DOM or store mutable values without triggering re-renders. 

**useState**
useState is used for managing component state, triggering re-renders when the state updates.

I'm using these values, and I want the most up-to-date information the user has changed.  I'd rather use useState here.

Reply:
with onValueChange={(val) => console.log(val)}, it returns current value without using useState. and NextUI already takes care of it and rerender it. if you pass the useRef variable to the ref attribute, it will update the ref when the value changes. That's why you can still type and see the value changes without passing any useState to the NextUI component like just this `<Input />`

Reply:
Is there a reason you don't want to use useState? Why is it essential to use useRef here?

Reply:
not essential, but prevents unneccessary rerendering, and you are using multiple usestates, so if a user types something in the definition textbox, it will rerender this whole component including other textboxes. it is optimal to rerender the definition textbox only.

Reply:
https://github.com/rainawan/BruinDictionary/assets/96547645/fb8196d3-7265-4b9b-aeec-7214244f1772

I am using a dev tool to check which components are rendered by highlighting the rerendered component.

Reply:
and your code would look cleaner cuz you don't need all the handlechange functions

Reply:
https://github.com/rainawan/BruinDictionary/assets/96547645/87632af1-082b-435e-95b0-f17548ebee71

if you use useref, it will rerender the corresponding textbox you are typing

Reply:
I'm still not sold on using useRef in this case.

From the docs:

"Changing a ref does not trigger a re-render, so refs are not appropriate for storing information you want to display on the screen. Use state for that instead."

This is what I want to do. I want to store information while simultaneously displaying it to the user correctly.

I know it's cleaner code, and your plugin shows what it shows, but I don't think that is how useRef is meant to be used. We would basically be trying to make useRef stand in for useState. Also, in the callback, you showed that you could access the values, but I am still not convinced that is the most up-to-date information. It looks like it is, but I have had projects where the information is a milli-second old and wrong while the app State has the updated info.

https://react.dev/reference/react/useRef

Reply:
i might be wrong on the use case of useRef, cuz I haven't used it so much. but we cares about the most recent input when the submit button is clicked and the rest rerendering will be handled by the component. then we can do ref.focus() when the submit button is clicked?

Reply:
https://youtu.be/w89kn-0E2CA?si=yKMcsEtifhe6EOWc This guy is doing what I mentioned

Reply:
One more, you are not even printing the state values on the screen like `<p>{definetext}</p>`and why do we need useState

Reply:
Ehhh. I'll think about it.

Reply:
done

Reply:
use ref added